src/ocr_extractor/storage.py

```diff
@@ -0,0 +1,164 @@
+import os
+import io
+import logging
+from typing import Optional
+from PIL import Image
+import boto3
+from botocore.client import Config
+from botocore.exceptions import ClientError
+
+logging.getLogger().setLevel(logging.INFO)
+
+class StorageHandler:
+    """ Store images / files in the s3 or local disk """
+    def __init__(
+        self,
+        use_s3: bool,
+        bucket_name: str,
+        bucket_key: str,
+        aws_region_name: str
+    ) -> None:
+        self.use_s3 = use_s3
+        self.bucket_name = bucket_name
+        self.bucket_key = bucket_key
+        self.s3_client = boto3.client(
+            "s3",
+            region_name=aws_region_name,
+            config=Config(
+                signature_version="s3v4",
+                s3={"addressing_style": "path"}
+            )
+        )
+
+    def generate_presigned_url(
+        self,
+        bucket_key: str,
+        signed_url_expiry_secs: int=86400
+    ) -> Optional[str]:
+        """
+        Generates a presigned url of the file stored in s3
+        """
+        try:
+            url = self.s3_client.generate_presigned_url(
+                ClientMethod="get_object",
+                Params={
+                    "Bucket": self.bucket_name,
+                    "Key": bucket_key
+                },
+                ExpiresIn=signed_url_expiry_secs
+            )
+        except ClientError as cexc:
+            logging.error("Error while generating presigned url %s", cexc)
+            return None
+        return url
+
+    def get_s3_link_or_local_path_for_image(
+        self,
+        image_data,
+        file_extension,
+        filename: str,
+        dir_type: str
+    ):
+        """ Store image file in s3 or local disk and returns the path to that file """
+        mapper = {
+            ".png": "image/png",
+            ".jpg": "image/jpg",
+            ".jpeg": "image/jpg",
+            ".gif": "image/gif"
+        }
+        file_extension = file_extension.group() if file_extension else ".jpg"
+        file_ext = mapper.get(file_extension, "image/jpg") if file_extension else "image/jpg"
```

Comment:
```suggestion
        file_ext = mapper.get(file_extension, "image/jpg")
```
we dont need if condition here